app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt

```diff
@@ -101,15 +99,18 @@ class LeanplumMetricsService(
 
     @Suppress("ComplexMethod")
     override fun start() {
+        val preferencesLoaded = scope.async {
+                preferences = application.getSharedPreferences(PREFERENCE_NAME, MODE_PRIVATE)
+        }
 
         if (!application.settings().isMarketingTelemetryEnabled) return
 
-        Leanplum.setIsTestModeEnabled(false)
-        Leanplum.setApplicationContext(application)
-        Leanplum.setDeviceId(deviceId)
-        Parser.parseVariables(application)
-
         leanplumJob = scope.launch {
+            preferencesLoaded.await()
+            Leanplum.setIsTestModeEnabled(false)
+            Leanplum.setApplicationContext(application)
+            Leanplum.setDeviceId(deviceId)
+            Parser.parseVariables(application)
```

Comment:
I can't spend enough time on this to know if this is thread safe. It's not documented as thread safe but many of the components it uses are synchronized. However, like a lot of Leanplum, none of the fields are documented as `volatile`, which makes me think it's technically not thread safe but we're treating it as such anyway.

However, from a theory perspective, I think this is safe (minus the volatile issue): leanplum hasn't started yet so no other code should be running to cause concurrency issues. It could be a problem if `LeanplumMetricsSerivce.start` is called twice without a `stop` in between, however. But this shouldn't happen because it's started in the Activity.

Also, we don't add any Leanplum annotations (e.g. `@Variable`) so in the current code, this method will exit early. That's good for now but it might create issues if we later do use other annotations.

@boek What do you think? Is it safe to put `Parser.parseVariables` on a background thread?